### ISSUE TITLE
pull request sunos update suggestions

### ISSUE THREAD

**Issue by risto3** [LudovicRousseau/PCSC#6]
can't seem to do a pull request on this repo, but here is the diff:
https://github.com/LudovicRousseau/PCSC/compare/master...risto3:master

SunOS traditionally doesn't have d_type in dirent
should cast to int avoiding warnings with pid_t 
and needs alloca.h for alloca()

**Comment by LudovicRousseau** [comment]
You can create Pull Requests.

Can you add the compiler messages you get with the unpatched code?
I do not like to change code if I do no understand what the problem is.

A documented PR for each of the 3 independent changes would be perfect.

**Comment by risto3** [comment]
a PR here?

**Comment by LudovicRousseau** [comment]
A PR for the PCSC repository, yes.
But not here in the "Issues".

**Comment by risto3** [comment]
The following are the compiler issues in two passes:
```
  CC       pcscd-dyn_macosx.o
configfile.l: In function 'DBGetReaderListDir':
configfile.l:293:16: error: 'struct dirent' has no member named 'd_type'
    if (direntry->d_type == DT_UNKNOWN)
                ^
configfile.l:293:28: error: 'DT_UNKNOWN' undeclared (first use in this function)
    if (direntry->d_type == DT_UNKNOWN)
                            ^
configfile.l:293:28: note: each undeclared identifier is reported only once for each function it appears in
configfile.l:312:17: error: 'struct dirent' has no member named 'd_type'
     if (direntry->d_type != DT_REG)
                 ^
configfile.l:312:29: error: 'DT_REG' undeclared (first use in this function)
     if (direntry->d_type != DT_REG)
                             ^
  CC       pcscd-dyn_unix.o
Makefile:883: recipe for target 'pcscd-configfile.o' failed
gmake[3]: *** [pcscd-configfile.o] Error 1
gmake[3]: *** Waiting for unfinished jobs....
utils.c: In function 'SendHotplugSignal':
utils.c:97:3: warning: format '%d' expects argument of type 'int', but argument 6 has type 'pid_t' [-Wformat=]
   Log2(PCSC_LOG_INFO, "Send hotplug signal to pcscd (pid=%d)", pid);
   ^
utils.c:100:4: warning: format '%d' expects argument of type 'int', but argument 6 has type 'pid_t' [-Wformat=]
    Log3(PCSC_LOG_CRITICAL, "Can't signal pcscd (pid=%d): %s",
    ^
```
and

```
  CC       pcscd-prothandler.o
pcscdaemon.c: In function 'main':
pcscdaemon.c:402:5: warning: format '%d' expects argument of type 'int', but argument 6 has type 'pid_t' [-Wformat=]
     Log2(PCSC_LOG_CRITICAL,
     ^
  CC       pcscd-readerfactory.o
  CC       pcscd-sd-daemon.o
  CC       pcscd-simclist.o
  CC       pcscd-sys_unix.o
  CC       pcscd-tokenparser.o
  CC       pcscd-hotplug_libudev.o
readerfactory.c: In function 'RFAddReader':
readerfactory.c:211:2: warning: implicit declaration of function 'alloca' [-Wimplicit-function-declaration]
  readerName = alloca(strlen(readerNameLong)+1);
  ^
readerfactory.c:211:15: warning: incompatible implicit declaration of built-in function 'alloca'
  readerName = alloca(strlen(readerNameLong)+1);
               ^
  CC       pcscd-hotplug_libusb.o
```

**Comment by risto3** [comment]
BTW, there is an autoconf test AC_FUNC_ALLOCA for alloca() and one as well called AC_STRUCT_DIRENT_D_TYPE for availability of d_type in dirent.

**Comment by LudovicRousseau** [comment]
Fixed upstream.
Thanks

**Comment by risto3** [comment]
unfortunately the second guard in configfile.l is misplaced.
it needs to hide the complete else statement...
I'll shoot another pull request.
